src/osd/PG.cc

```diff
@@ -5316,7 +5316,15 @@ void PG::on_new_interval()
     upacting_features &= osdmap->get_xinfo(*p).features;
   }
 
-  assert(osdmap->test_flag(CEPH_OSDMAP_SORTBITWISE));
+  // we don't speak non-sortbitwise, so we'd better not go
+  // active if they somehow haven't set it!
+  if (!osdmap->test_flag(CEPH_OSDMAP_SORTBITWISE) ||
+	 // but we have to be nice in case we're sortbitwise *now*
+	 // and are processing updates from before that happened
+      (osd->osd->is_active() && osdmap->get_epoch() < osd->get_boot_epoch())) {
```

Comment:
In my first pass at this I did just drop the PG assert. I ended up not doing that because both this and the OSD::activate_map() one were inserted as part of one PR from you, and they're accomplishing slightly different things — the broader OSD one is called *after* the OSDMap gets published, which I think means it can leak out to PGs and get used before the main OSD manages to shut everything down. Not sure if there are other invariants they're covering or not.